src/domain/discounts/new/index.js

```diff
@@ -131,31 +143,41 @@ const NewDiscount = ({}) => {
 
     data.rule.value = parseInt(data.rule.value)
 
-    if (data.rule.type === "fixed") {
+    if (!isPercentageDiscount) {
       data.rule.value = data.rule.value * 100
     }
+    data.rule.type = isPercentageDiscount ? "percentage" : "fixed"
+    data.rule.allocation = isAllocatedToItem ? "item" : "total"
 
     data.rule.valid_for = validProducts()
     data.regions = validRegions()
 
     const discount = {
       code: data.code,
-      is_dynamic: data.is_dynamic === "true",
+      is_dynamic: isDynamic, //data.is_dynamic === "true",
       rule: data.rule,
+      starts_at: moment(startDate).format("MM/DD/YYYY HH:mm"),
+      ends_at: endDate ? moment(endDate).format("MM/DD/YYYY HH:mm") : undefined,
       regions: data.regions || [],
+      valid_duration: isDynamic ? iso8601Date : undefined,
     }
 
     if (data.usage_limit) {
       discount.usage_limit = data.usage_limit
     }
 
+    console.log(discount)
```

Comment:
Remove this :)

Reply:
yes, that seems redundant, I have removed it 😄